KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,714 @@
+    event NewbieRegistered(address newbie, uint256 fundAllocation);
+    event NewbieRemoved(address newbie, uint256 newbieCount);
+    event RetiredAddrAdminStatus(bool success, bytes result);
+    event Approved(address retired, address approver, uint256 approversCount);
+    event StatusChanged(Status status);
+    event Finalized(string memo, Status status);
+
+    /**
+     * Modifiers
+     */
+    modifier onlyAtStatus(Status _status) {
+        require(status == _status, "Not in the designated status");
+        _;
+    }
+
+    /**
+     *  Constructor
+     * @param _rebalanceBlockNumber is the target block number of the execution the rebalance in Core
+     */
+    constructor(uint256 _rebalanceBlockNumber) {
+        rebalanceBlockNumber = _rebalanceBlockNumber;
+        status = Status.Initialized;
+        emit ContractDeployed(status, _rebalanceBlockNumber, block.timestamp);
+    }
+
+    //State changing Functions
+    /**
+     * @dev registers retired details
+     * @param _retiredAddress is the address of the retired
+     */
+    function registerRetired(
+        address _retiredAddress
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        require(
+            !retiredExists(_retiredAddress),
+            "Retired address is already registered"
+        );
+        Retired storage retired = retirees.push();
+        retired.retired = _retiredAddress;
+        emit RetiredRegistered(retired.retired, retired.approvers);
+    }
+
+    /**
+     * @dev remove the retired details from the array
+     * @param _retiredAddress is the address of the retired
+     */
+    function removeRetired(
+        address _retiredAddress
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        uint256 retiredIndex = getRetiredIndex(_retiredAddress);
+        retirees[retiredIndex] = retirees[retirees.length - 1];
+        retirees.pop();
+
+        emit RetiredRemoved(_retiredAddress, retirees.length);
+    }
+
+    /**
+     * @dev registers newbie address and its fund distribution
+     * @param _newbieAddress is the address of the newbie
+     * @param _amount is the fund to be allocated to the newbie
+     */
+    function registerNewbie(
+        address _newbieAddress,
+        uint256 _amount
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        require(
+            !newbieExists(_newbieAddress),
+            "Newbie address is already registered"
+        );
+        require(_amount != 0, "Amount cannot be set to 0");
+
+        Newbie memory newbie = Newbie(_newbieAddress, _amount);
+        newbies.push(newbie);
+
+        emit NewbieRegistered(_newbieAddress, _amount);
+    }
+
+    /**
+     * @dev remove the newbie details from the array
+     * @param _newbieAddress is the address of the newbie
+     */
+    function removeNewbie(
+        address _newbieAddress
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        uint256 newbieIndex = getNewbieIndex(_newbieAddress);
```

Comment:
need to check if newbieIndex == type(uint256).max

Reply:
added checks